Check `package.json` health

mxdvl opened the pull request:
## What does this change?

- all dependencies are development dependencies
- missing peer dependencies are specified
- added list of known issues
- enforce a healthy package.json with https://github.com/guardian/actions-npm-dependencies/

## Why?

We have started to see the issues related to dependencies:
- missing peer dependencies in #6945 and #6946 
- `@types/*` dependencies in #6971

## Screenshots

<img width="830" alt="image" src="https://user-images.githubusercontent.com/76776/216010845-2c9cb937-c4a7-4a72-90b5-b14eb4bdfce7.png">

See full run: https://github.com/guardian/dotcom-rendering/actions/runs/4063074923/jobs/6994876002

bryophyta commented:
@mxdvl it looks like the CI checks for this aren't passing at the moment and the branch hasn't been updated for a couple of weeks, so I'm going to add the `prnouncer-ignore` label to the PR. Please do remove the label whenever you want it to be reviewed though!

p.s. this looks like a really interesting project, I'm looking forward to reviewing it!

mxdvl commented:
> @mxdvl it looks like the CI checks for this aren't passing at the moment and the branch hasn't been updated for a couple of weeks, so I'm going to add the `prnouncer-ignore` label to the PR. Please do remove the label whenever you want it to be reviewed though!
> 
> p.s. this looks like a really interesting project, I'm looking forward to reviewing it!

I’ve rebased and fixed all the other tests, but there are some remaining issues with the actual checking. I don’t have too much bandwith to fix this at the minute, so happy to close if there’s not much interest from the team.

bryophyta commented:
> I’ve rebased and fixed all the other tests, but there are some remaining issues with the actual checking. I don’t have too much bandwith to fix this at the minute, so happy to close if there’s not much interest from the team.

Thanks for the update! Fwiw it seems like a really worthwhile check so I'd be very happy for it to be included at some point! I don't want to put more work on your plate though; is it something that would be useful to pair on at some point?

mxdvl commented:
There was a suggestion that this would be better suited in a place like CSNX
